Utils/Services/ImageFetcher.swift

```diff
@@ -11,6 +11,12 @@ import ReactiveSwift
 
 public protocol ImageFetcherType {
     
+    /**
+     Fetches an image asynchronously from a URL, returning a value or an error in the SignalProducer when done.
+     - parameter imageURL: the url of the image to fetch.
+     - parameter session: the URLSession to use to make the request. Default: URLSession.shared.
```

Comment:
Also (can't write it in the correct line) `ImageFetcherError` misses documentation. Maybe just a small 
```
/**
Enum that represents possible errors from trying to fetch an image.
It includes:
    a successful fetch but whose result file is not an image <if there are certain formats that are understood it would be nice to put a link to where apple says these formats>, or
    a problem in the communication with an Error associated that gives more information (this may include no internet connection or inexistent URL, <etc>).
```